### DISCUSSION TITLE
Add support for XDG base directory specification

### DISCUSSION THREAD

**Discussion by Disonantemus** [d99kris/nchat#171] [Ideas]
Support [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html), to have an alternative 4 nchat folder to be inside `~/.config`

I think is better, to not populate too much **home** (~), with files and folders.

**Comment by d99kris** [comment]
Thanks for submitting the request. This suggestion has come up before - maybe in https://t.me/nchatusers - but has dropped off my radar. I agree it's a reasonable functionality  and it should be fairly straightforward to implement. I'll try to look into it when I find some time.

**Comment by Kabouik** [comment]
I agree and I actually circumvent that by setting up aliases for my different accounts that point to the correct account folders in `~/.config/nchat/xxx` (and same with `nmail`), but given that they may both fetch a lot of data, it would be nice if they could store the data in `~/.local/nchat/xxx`  and `~/.local/nmail/xxx` too and just keep the configuration files in `~/.config`, which I cannot do with aliases only.

**Comment by d99kris** [comment]
Basic XDG directory spec support has been added now. Thanks to @torrca for contributing with the implementation in PR #211.
For now it only switches default directory to `~/.config/nchat` (or based on XDG_CONFIG_HOME if set), so no separation of config and data yet. But it's a start. To be backward compatible it uses `~/.nchat` if found.

**Comment by d99kris** [comment]
I will proceed to close this feature request. If it's desired to have separate config and data storage, please report a new feature request for that, to its interest can be gauged separately from the home dir pollution issue. Thanks!
